R/assess_dependencies.R

```diff
@@ -0,0 +1,121 @@
+#' Assessment of dependency footprint for a specific package
+#'
+#' Only Depends, Imports and LinkingTo dependencies are assessed because
+#' they are required
+#'
+#' @details The more packages a package relies on the more chances for errors exist.
+#'
+#' @eval roxygen_assess_family(
+#'   "dependencies",
+#'   "a dataframe of package names and they type of dependency the package being assess has to them")
+#'
+#'
+#' @export
+assess_dependencies <- function(x, ...){
+  UseMethod("assess_dependencies")
+}
+
+attributes(assess_dependencies)$column_name <- "dependencies"
+attributes(assess_dependencies)$label <- "Package dependency footprint"
+
+#' @export
+assess_dependencies.default <- function(x, ...){
+  as_pkg_metric_na(pkg_metric(class = "pkg_metric_dependencies"))
+}
+
+#' @export
+assess_dependencies.pkg_source <- function(x, ...){
+  pkg_metric_eval(class = "pkg_metric_dependencies", {
+    parse_dcf_dependencies(x$path)
+    })
+}
+
+#' @export
+assess_dependencies.pkg_install <- function(x, ...){
+  pkg_metric_eval(class = "pkg_metric_dependencies", {
+    parse_dcf_dependencies(x$path)
+  })
+}
+
+#' @export
+assess_dependencies.pkg_cran_remote <- function(x, ...){
+  pkg_metric_eval(class = "pkg_metric_dependencies", {
+      get_package_dependencies(x$name, repo = getOption("repos")[["CRAN"]])
```

Comment:
I don't think we necessarily need to change this before pulling in changes, but we may also want to consider unnamed repos that match a repo listed in `utils::getCRANmirrors()` (we already cache this with `memoise_cran_mirrors()`) so that we don't require that `getOption("repos")` always have names.

Reply:
incorporated this suggestion, ended up taking it one step further by checking against memoise_cran_mirrors, if that returns empty then checking all urls in getOptions("repos") for a "cran" subdomain and finally for a named "CRAN" repo